### PR TITLE
Rewrite relations query with goqu

### PR DESCRIPTION
**Pull request by SherinV**
Signed-off-by: Sherin Varughese <shvarugh@redhat.com>

For https://github.com/stolostron/backlog/issues/20282
Rewrite relations query with goqu template
~Add relatedKinds filter~

### CODE REVIEW

File: pkg/config/config.go
Comment (jlpadilla) [inline pkg/config/config.go:7]: Thanks for making this consistent with indexer!

File: pkg/resolver/search.go
Comment (jlpadilla) [inline pkg/resolver/search.go:201]: Should we keep a comment with the original query for documentation?

File: pkg/resolver/search.go
Comment (jlpadilla) [inline pkg/resolver/search.go:235]: I wonder if we are able to continue after this error.

File: pkg/resolver/search.go
Comment (jlpadilla) [inline pkg/resolver/search.go:237]: context.TODO()

File: pkg/resolver/search.go
Comment (SherinV) [reply pkg/resolver/search.go:201]: Done

File: pkg/resolver/search.go
Comment (jlpadilla) [reply pkg/resolver/search.go:255]: Would the client receive an error? GraphQL can send partial results and errors.
Also the error we send to the client shouldn't describe any internals like the SQL query.

**We can move this discussion out of this PR, we should review the error path for the entire project.**

Comment (SherinV) [reply pkg/resolver/search.go:256]: Yes, we will revisit the error path for the entire api along with the context.

**Review by jlpadilla — APPROVED** [review]
/lgtm
/hold

**Pull request merged**
